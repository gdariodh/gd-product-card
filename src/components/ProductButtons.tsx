import React,{ CSSProperties, useCallback, useContext } from "react";
import { ProductContext } from "./ProductCard";
import styles from "../styles/styles.module.css";

export interface Props {
  className?: string;
  activeBtnClass?: string;
  style?: CSSProperties;
}

export const ProductButtons = ({ className, style }: Props) => {
  const { counter, increaseBy, maxCount } = useContext(ProductContext);

  // devuelve true o false, comprobamos si existe maxCount, y si es igual al counter
  const isMaxReached = useCallback(
    () => !!maxCount && counter === maxCount,
    [counter, maxCount]
  );

  // TODO: NOTA -> !! sirve para negar es decir si es undefined le aplica el !! y lo vuelve false
  // y asi lo mantenemos true o false

  return (
    <div className={`${styles.buttonsContainer} ${className}`} style={style}>
      <button className={styles.buttonMinus} onClick={() => increaseBy(-1)}>
        -
      </button>

      <div className={styles.countLabel}>{counter}</div>

      <button
        className={`${styles.buttonAdd} ${isMaxReached() && styles.disabled}`}
        onClick={() => increaseBy(+1)}
      >
        +
      </button>
    </div>
  );
};
