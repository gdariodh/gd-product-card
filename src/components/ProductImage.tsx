import React, { CSSProperties, useContext } from "react";
import { ProductContext } from "./ProductCard";
import styles from "../styles/styles.module.css";
import noImage from "../assets/no-image.jpg"; // acceder a img q no esta en public

export interface Props {
  className?: string;
  img?: string;
  style?: CSSProperties;
}

export const ProductImage = ({ img, className, style }: Props) => {
  const { product } = useContext(ProductContext);

  let imgToShow: string;

  if (img) {
    imgToShow = img;
  } else if (product.img) {
    imgToShow = product.img;
  } else {
    imgToShow = noImage;
  }

  return (
    <img
      src={imgToShow}
      alt="Product Image"
      className={`${styles.productImg} ${className}`}
      style={style}
    />
  );
};
