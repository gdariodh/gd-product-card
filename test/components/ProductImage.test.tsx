import React from 'react';
import renderer from 'react-test-renderer';
import { ProductImage, ProductCard } from '../../src/components';
import { product2 } from '../data/products';

describe('ProductImage', () => {
  
    test('ProductImage debe mostrar el componente con la imagen', () => {
        const wrapper = renderer.create(
            <ProductImage img={product2.img} className="custom-class"/>
        );

        expect(wrapper.toJSON()).toMatchSnapshot()
    });

    test('debe de mostrar el componente con la imagen pero con context', () => {
      
        const wrapper = renderer.create(
            <ProductCard product={product2}>
                {
                    () => (
                        <ProductImage/>
                    )
                }
            </ProductCard>
        );

        expect(wrapper.toJSON()).toMatchSnapshot()
    })
})