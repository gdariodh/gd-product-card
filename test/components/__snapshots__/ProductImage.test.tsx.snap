// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage ProductImage debe mostrar el componente con la imagen 1`] = `
<img
  alt="Product Image"
  className="productImg custom-class"
  src="./images/coffee-mug.png"
/>
`;

exports[`ProductImage debe de mostrar el componente con la imagen pero con context 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product Image"
    className="productImg undefined"
    src="./images/coffee-mug.png"
  />
</div>
`;
